app/src/main/java/com/example/groceriesmanager/Activities/EditFoodItemActivity.java

```diff
@@ -86,26 +87,28 @@ public void onClick(View v) {
                 String foodQty = etFoodQty.getText().toString();
 
                 String foodMeasure = spinnerFoodMeasure.getSelectedItem().toString();
-                // Toast.makeText(AddFoodItemActivity.this, foodName +": " + String.valueOf(foodQty) + " " + foodMeasure, Toast.LENGTH_LONG).show();
+                // todo: fix error here. get selected item
+                String foodCategory = Arrays.asList(getResources().getStringArray(R.array.food_categories)).get(spinnerFoodCategory.getSelectedItemPosition());
+                Log.i(TAG, "food category selection: " + foodCategory);
+
 
                 if (foodName.replaceAll("\\s+", "").length()==0){ // if the user did not type in a food name or types only spaces
                     Toast.makeText(EditFoodItemActivity.this, "type in the food name", Toast.LENGTH_LONG).show();
 
                 }
                 else{
-                    if (Objects.equals(process, "new")){
-                        addFoodItem(foodName, foodQty, foodMeasure, type);
+                    if (Objects.equals(process, "new")){ // is user is creating new food item
+                        addFoodItem(foodName, foodQty, foodMeasure, type, foodCategory);
                     }
                     else{ // process is edit
-                        // todo: populate. if process is edit,
-                        editFoodItem(foodName, foodQty, foodMeasure, type);
+                        editFoodItem(foodName, foodQty, foodMeasure, type, foodCategory);
                     }
                 }
         }
         });
     }
 
-    private void editFoodItem(String foodName, String foodQty, String foodMeasure, String type) {
```

Comment:
I noticed the same change for the addFoodItem as well, perhaps a better approach for this would be to create a foodItem struct or class that contains this information and pass that to these functions instead.  It would make further modifications and changes easier to work with, and the function signature would not need to keep changing.